combination of Neo4j 4.3.3 and Redis dependencies causes Neo4j to crash on start (Linux only)

## Guidelines

Please note that GitHub issues are only meant for bug reports/feature requests. If you have questions on how to use APOC, please ask on [the Neo4j Discussion Forum](https://community.neo4j.com/c/neo4j-graph-platform/procedures-apoc) instead of creating an issue here.

## Expected Behavior (Mandatory)
Neo4j 4.3.4 starts in console mode with all APOC 4.3.0.1 dependency jars in plugins folder

## Actual Behavior (Mandatory)
Above works on Windows, fails on Linux (Ubuntu 20.04)
Removing the **redis** jar solves the issue, the error message also indicates that's where the issue is.

### Steps (Mandatory)
```
neo4j-enterprise-4.3.4/bin/neo4j console
Directories in use:
home:         /home/tomgeu/neo4j/neo4j-enterprise-4.3.4
config:       /home/tomgeu/neo4j/neo4j-enterprise-4.3.4/conf
logs:         /home/tomgeu/neo4j/neo4j-enterprise-4.3.4/logs
plugins:      /home/tomgeu/neo4j/neo4j-enterprise-4.3.4/plugins
import:       /home/tomgeu/neo4j/neo4j-enterprise-4.3.4/import
data:         /home/tomgeu/neo4j/neo4j-enterprise-4.3.4/data
certificates: /home/tomgeu/neo4j/neo4j-enterprise-4.3.4/certificates
licenses:     /home/tomgeu/neo4j/neo4j-enterprise-4.3.4/licenses
run:          /home/tomgeu/neo4j/neo4j-enterprise-4.3.4/run
Starting Neo4j.
WARNING: Max 4096 open files allowed, minimum of 40000 recommended. See the Neo4j manual.
2021-09-23 11:06:37.993+0000 INFO  Starting...
2021-09-23 11:06:39.323+0000 INFO  ======== Neo4j 4.3.4 ========
2021-09-23 11:06:41.119+0000 INFO  Called db.clearQueryCaches(): Query cache already empty.
2021-09-23 11:07:16.539+0000 ERROR Failed to start Neo4j on dbms.connector.http.listen_address, a socket address. If missing port or hostname it is acquired from dbms.default_listen_address.
java.lang.RuntimeException: Error starting Neo4j database server at /home/tomgeu/neo4j/neo4j-enterprise-4.3.4/data/databases
        at org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.startDatabaseServer(DatabaseManagementServiceFactory.java:214) ~[neo4j-4.3.4.jar:4.3.4]
        at org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.build(DatabaseManagementServiceFactory.java:170) ~[neo4j-4.3.4.jar:4.3.4]
        at com.neo4j.server.enterprise.EnterpriseManagementServiceFactory.createManagementService(EnterpriseManagementServiceFactory.java:38) ~[neo4j-enterprise-4.3.4.jar:4.3.4]
        at com.neo4j.server.enterprise.EnterpriseBootstrapper.createNeo(EnterpriseBootstrapper.java:20) ~[neo4j-enterprise-4.3.4.jar:4.3.4]
        at org.neo4j.server.NeoBootstrapper.start(NeoBootstrapper.java:134) [neo4j-4.3.4.jar:4.3.4]
        at org.neo4j.server.NeoBootstrapper.start(NeoBootstrapper.java:90) [neo4j-4.3.4.jar:4.3.4]
        at com.neo4j.server.enterprise.EnterpriseEntryPoint.main(EnterpriseEntryPoint.java:24) [neo4j-enterprise-4.3.4.jar:4.3.4]
Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'Server[backup-server]' was successfully initialized, but failed to start. Please see the attached cause exception "Could not initialize class io.netty.channel.epoll.EpollChannelOption".
        at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:463) ~[neo4j-common-4.3.4.jar:4.3.4]
        at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:110) ~[neo4j-common-4.3.4.jar:4.3.4]
        at org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.startDatabaseServer(DatabaseManagementServiceFactory.java:205) ~[neo4j-4.3.4.jar:4.3.4]
        ... 6 more
Caused by: java.lang.NoClassDefFoundError: Could not initialize class io.netty.channel.epoll.EpollChannelOption
        at io.netty.channel.epoll.EpollServerSocketChannelConfig.getOption(EpollServerSocketChannelConfig.java:54) ~[netty-all-4.1.68.Final.jar:4.1.68.Final]
        at io.netty.channel.AbstractChannel$AbstractUnsafe.bind(AbstractChannel.java:548) ~[apoc-redis-dependencies-4.3.0.1.jar:4.3.0.1]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.bind(DefaultChannelPipeline.java:1334) ~[apoc-redis-dependencies-4.3.0.1.jar:4.3.0.1]
        at io.netty.channel.AbstractChannelHandlerContext.invokeBind(AbstractChannelHandlerContext.java:506) ~[apoc-redis-dependencies-4.3.0.1.jar:4.3.0.1]
        at io.netty.channel.AbstractChannelHandlerContext.bind(AbstractChannelHandlerContext.java:491) ~[apoc-redis-dependencies-4.3.0.1.jar:4.3.0.1]
        at io.netty.channel.DefaultChannelPipeline.bind(DefaultChannelPipeline.java:973) ~[apoc-redis-dependencies-4.3.0.1.jar:4.3.0.1]
        at io.netty.channel.AbstractChannel.bind(AbstractChannel.java:260) ~[apoc-redis-dependencies-4.3.0.1.jar:4.3.0.1]
        at io.netty.bootstrap.AbstractBootstrap$2.run(AbstractBootstrap.java:356) ~[apoc-redis-dependencies-4.3.0.1.jar:4.3.0.1]
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164) ~[apoc-redis-dependencies-4.3.0.1.jar:4.3.0.1]
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472) ~[apoc-redis-dependencies-4.3.0.1.jar:4.3.0.1]
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:387) ~[netty-all-4.1.68.Final.jar:4.1.68.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[apoc-redis-dependencies-4.3.0.1.jar:4.3.0.1]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[apoc-redis-dependencies-4.3.0.1.jar:4.3.0.1]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:?]
        at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
        at java.lang.Thread.run(Unknown Source) ~[?:?]
2021-09-23 11:07:16.541+0000 INFO  Neo4j Server shutdown initiated by request
2021-09-23 11:07:16.541+0000 INFO  Stopped.
Unexpected process failure. See log for info.
Run with '--verbose' for a more detailed error message.
```

## Specifications (Mandatory)
### Versions

  - OS: Ubuntu 20.04
  - Neo4j: 4.3.4
  - Neo4j-Apoc: 4.3.0.1

--- a/docs/asciidoc/modules/ROOT/pages/database-integration/redis.adoc
+++ b/docs/asciidoc/modules/ROOT/pages/database-integration/redis.adoc
@@ -44,7 +44,7 @@ Here is a list of all available Redis procedures:
 == Install Dependencies
 
 The Redis procedures have dependencies on a client library that is not included in the APOC Library.
-You can download it from https://github.com/lettuce-io/lettuce-core/releases/tag/6.1.1.RELEASE[the lettuce-core repository] 
+You can download it from https://github.com/lettuce-io/lettuce-core/releases/tag/6.1.1.RELEASE[the lettuce-core repository](except for `netty` jars because they are already included within neo4j)
 or https://github.com/neo4j-contrib/neo4j-apoc-procedures/releases/download/{apoc-release}/apoc-redis-dependencies-{apoc-release}.jar[apoc repository]
 Once that file is downloaded, it should be placed in the `plugins` directory and the Neo4j Server restarted.
 

--- a/extra-dependencies/redis/build.gradle
+++ b/extra-dependencies/redis/build.gradle
@@ -17,5 +17,7 @@ jar {
 }
 
 dependencies {
-    compile group: 'io.lettuce', name: 'lettuce-core', version: '6.1.1.RELEASE'
+    compile group: 'io.lettuce', name: 'lettuce-core', version: '6.1.1.RELEASE', {
+        exclude group: 'io.netty'
+    }
 }

